### DISCUSSION TITLE
Integration with Partytown

### DISCUSSION THREAD

**Discussion by franzus5** [johannschopplich/nuxt-gtag#25] [Q&A] [answered]
First of all, thank you for your effort!

**The Problem:**
So the problem that we're faced is that in [pagespeed.web.dev](https://pagespeed.web.dev/) we get the error of GTM (usual problem) which decreases the LPC vitals.

[Screen](https://monosnap.com/file/RwEtQerL5N8D1lGItqKquFShBHF1od)

**Ways to solve:**
Like easy option is using `Partytown` lib to make 3d part libs load after the main thread. So the idea is to add the possibility of an optional `type` to settings https://github.com/johannschopplich/nuxt-gtag/blob/main/src/runtime/plugin.client.ts#L26 to define this import like 3d part for Partytown.

```ts
useHead({
    script: [
      {
        src: `https://www.googletagmanager.com/gtag/js?id=${id}`,
        [strategy]: true,
        ...type // type: 'text/partytown'
      },
    ],
  })
```

Or maybe you have other thoughts on how we can proceed with this, would be very useful. Thank you!

**Comment by johannschopplich** [comment, marked as answer]
Sure, you can use Partytown. Actually, I'd recommend that approach over this module. For the Partytown implementation, you don't even need `nuxt-gtag`:

```ts
useHead({
  script: [
    {
      src: `https://www.googletagmanager.com/gtag/js?id=${process.env.NUXT_PUBLIC_GTAG_ID}`,
      type: 'text/partytown'
    },
  ],
})
```

**Reply by franzus5** [reply]
Thank you!
